component-samples/scripts/keys_gen.sh

```diff
@@ -0,0 +1,247 @@
+                -name ${1}_rsa2048 -out $CERTS_PATH/${1}_rsa2048.p12 -password pass:${PASS_KEY}
+        openssl pkcs12 -export -in $CERTS_PATH/${1}_ec256_cert.pem -inkey $CERTS_PATH/${1}_ec256_private.key \
+                -name ${1}_ecdsa_256 -out $CERTS_PATH/${1}_ec256.p12 -password pass:${PASS_KEY}
+        openssl pkcs12 -export -in $CERTS_PATH/${1}_ec384_cert.pem -inkey $CERTS_PATH/${1}_ec384_private.key \
+                -name ${1}_ecdsa_384 -out $CERTS_PATH/${1}_ec384.p12 -password pass:${PASS_KEY}
+
+        # Import rsa2048, ec256, and ec384 p12 keystores to one p12 keystore
+        keytool -importkeystore -srckeystore $CERTS_PATH/${1}_rsa2048.p12 -srcstorepass ${PASS_KEY} \
+                -destkeystore $CERTS_PATH/${1}_keystore.p12 -deststorepass ${PASS_KEY} -deststoretype pkcs12 -noprompt
+        keytool -importkeystore -srckeystore $CERTS_PATH/${1}_ec256.p12 -srcstorepass ${PASS_KEY} \
+                -destkeystore $CERTS_PATH/${1}_keystore.p12 -deststorepass ${PASS_KEY} -deststoretype pkcs12 -noprompt
+        keytool -importkeystore -srckeystore $CERTS_PATH/${1}_ec384.p12 -srcstorepass ${PASS_KEY} \
+                -destkeystore $CERTS_PATH/${1}_keystore.p12 -deststorepass ${PASS_KEY} -deststoretype pkcs12 -noprompt
+
+        # Verify the keys from p12 keystore
+        keytool -list -keystore $CERTS_PATH/${1}_keystore.p12 -storepass ${PASS_KEY}
+
+        # Creating a keys.pem files
+        cat $CERTS_PATH/${1}_ec256_public.key | tee $CERTS_PATH/${1}_pub_keys.pem > /dev/null
+        cat $CERTS_PATH/${1}_ec384_public.key | tee -a $CERTS_PATH/${1}_pub_keys.pem > /dev/null
+        cat $CERTS_PATH/${1}_rsa2048_public.key | tee -a $CERTS_PATH/${1}_pub_keys.pem > /dev/null
+
+        echo "@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@" | \
+                tee $CERTS_PATH/${1}_keys.hash > /dev/null
+        echo "${1} key password is '${PASS_KEY}'" | tee -a $CERTS_PATH/${1}_keys.hash > /dev/null
+        echo "@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@" | \
+                tee -a $CERTS_PATH/${1}_keys.hash > /dev/null
+}
+
+hash_gen()
+{
+        # Hash generations
+        openssl x509 -in $CERTS_PATH/${1}_rsa2048_cert.pem -pubkey -noout | openssl enc -base64 -d \
+                > $CERTS_PATH/${1}_rsa2048.der
+        export RSA2048_HASH=`cat $CERTS_PATH/${1}_rsa2048.der | openssl dgst -sha256 | awk '/s/{print toupper($2)}'`
+        openssl x509 -in $CERTS_PATH/${1}_ec256_cert.pem -pubkey -noout | openssl enc -base64 -d \
+                > $CERTS_PATH/${1}_ec256.der
+        export EC256_HASH=`cat $CERTS_PATH/${1}_ec256.der | openssl dgst -sha256 | awk '/s/{print toupper($2)}'`
+        openssl x509 -in $CERTS_PATH/${1}_ec384_cert.pem -pubkey -noout | openssl enc -base64 -d \
+                > $CERTS_PATH/${1}_ec384.der
+        export EC384_HASH=`cat $CERTS_PATH/${1}_ec384.der | openssl dgst -sha256 | awk '/s/{print toupper($2)}'`
```

Comment:
Is this HASH generation compatible with RV allowlist configuration? If we add this HASH value to RV source code, does it successfully pass the specific public key?

Reply:
Validated this scenario earlier. Scenario works fine.